ring/ring.go

```diff
@@ -402,6 +405,14 @@ func (r *Ring) Get(key uint32, op Operation, bufDescs []InstanceDesc, bufHosts,
 	}, nil
 }
 
+// GetTokens implement ReadRing.
```

Comment:
I'm not sure if caller has any use for these tokens. Aren't `InstancesCount` or `GetAllHealthy` enough to check if ring is not empty?

(This method is also unsafe: it should create a copy and not expose internal data of `*Ring` to client, which can then modify it)

Reply:
> I'm not sure if caller has any use for these tokens. Aren't InstancesCount or GetAllHealthy enough to check if ring is not empty?

They don't check for the number of tokens :(

Reply:
> They don't check for the number of tokens :(

Do you want to protect against the situation when there are only instances with no tokens in the ring? Does that happen in your setup?